core/include/vecmem/utils/tuple.hpp

```diff
@@ -79,8 +80,8 @@ struct tuple<T, Ts...> {
         std::enable_if_t<
             vecmem::details::conjunction<
                 vecmem::details::negation<std::is_same<tuple<T, Ts...>, U>>,
-                std::is_constructible<T, U &&>,
-                std::is_constructible<Ts, Us &&>...>::value,
```

Comment:
Why are you removing the rvalue reference markers here?

Reply:
I was going back and forth with this one. :thinking: It's the same discussion from before. That especially in this constructor we use universal references. So most of the time we don't actually use rvalue references, bur rather lvalue ones.

Is `U` and `Us` not the correct type of reference as-is in this specific function? :thinking:

Reply:
I was trying to figure more out with the help of Google, but didn't get much further. :frowning: Since I found a couple of examples of people using `U&&` in similar situations, I went back to using that. Even if I'm still not entirely sure that that's absolutely the most correct. :thinking:

But at least this formalism allows both this project and [traccc](https://github.com/acts-project/traccc) to build everything successfully as much as I could test it. (As did the formalism without `&&`...) So let's use this one.

Reply:
Indeed `U&&` is the correct solution. `std::forward<T>` always returns `T&&` so there really is no valid reason for anything but `U&&` to be correct. :confused: